### DISCUSSION TITLE
Deleted event reappears

### DISCUSSION THREAD

**Discussion by tkuiCO2** [derekantrican/GAS-ICS-Sync#407] [Q&A] [answered]
I've successfully set up [GAS-ICS-Sync] to keep my iCloud calendar and Gmail Calendar in sync. It's been working smoothly for a long time. However, I'm now facing an annoying issue with a series event that I was invited to by a third party on my iCloud calendar. This series was synced to my Gmail Calendar. The third party has since canceled the series and removed it from my iCloud calendar, but it still appears on my Gmail Calendar and hasn't been automatically removed. I've tried manually deleting the series from the Gmail Calendar, but it keeps coming back again and again, and I can't seem to get rid of it. I've exhausted all options with this series, including transferring ownership, moving it to another calendar, and even changing the date to the 90s, but I'm still stuck with it. I'm wondering, where is it coming from if it doesn't exist on my iCloud calendar? I would greatly appreciate any help in removing it from syncing!

**Comment by jonas0b1011001** [comment]
Please download the source ics, open it in any text editor, search for the event title and share the full event definition (BEGIN:VEVENT to END:VEVENT)

**Reply by tkuiCO2** [reply]
Due to privacy concerns, I have substituted the names with parameters enclosed in %% brackets, while leaving the remaining content unchanged:

BEGIN:VEVENT
ATTENDEE;CN=%My Name%;CUTYPE=INDIVIDUAL;EMAIL=%My Email%;PART
 STAT=DECLINED;ROLE=REQ-PARTICIPANT:/aMTM4MzM2MDk1MDEzODMzNvFpikHtcl1ytxR
 BixhV94vM7ahtCo322CYh7HixO5rJ/principal/
ATTENDEE;CN=%3rd Party Name%;CUTYPE=INDIVIDUAL;EMAIL=%3rd Party Email%;PARTSTAT=ACCEPTED;ROLE=CHAIR:/aNTU0NTI1OTM5NTU0NTI1OUR7EYKt0WI7ZWmY
 saX_BXlLk_eY7pZXhqPhtJFIKyLq/principal/
CREATED:20231209T064904Z
DTEND;TZID=Asia/Jer:20231219T120000
DTSTAMP:20231227T133511Z
DTSTART;TZID=Asia/Jer:20231219T110000
LAST-MODIFIED:20231227T133509Z
ORGANIZER;CN=%3rd party name%;EMAIL=%3rd party email%:/aNTU0NTI1OT
 M5NTU0NTI1OUR7EYKt0WI7ZWmYsaX_BXlLk_eY7pZXhqPhtJFIKyLq/principal/
RRULE:FREQ=WEEKLY
SEQUENCE:0
SUMMARY:%Event Name% 
UID:B39B2EDE-29B1-4118-8ED1-B1966F9209D2
URL;VALUE=URI:
X-APPLE-CREATOR-IDENTITY:com.apple.mobilecal
X-APPLE-CREATOR-TEAM-IDENTITY:0000000000
END:VEVENT

**Reply by jonas0b1011001** [reply]
> ATTENDEE;CN=%My Name%;CUTYPE=INDIVIDUAL;EMAIL=%My Email%;PART
STAT=DECLINED

That's the only part hinting at the event being cancelled. However that's your icloud account declining participation and for the script there is no connection between the icloud account and the google account.

Check [this comment](https://github.com/derekantrican/GAS-ICS-Sync/discussions/368#discussioncomment-7162140) and see if that solves it.

**Reply by tkuiCO2** [reply]
It didn't help. I tried both snippets...
I have the event UID: B39B2EDE-29B1-4118-8ED1-B1966F9209D2.
Is there anything I can do with that?

**Reply by jonas0b1011001** [reply]
Just realized you have to change
`let mail = attendee.getParameter('cn');`
to
`let mail = attendee.getParameter('email');`

---
> I have the event UID: B39B2EDE-29B1-4118-8ED1-B1966F9209D2.
> Is there anything I can do with that?

If above still does not work replace the whole snippet with:
```
result = result.filter(function(event){
    try{
      return (event.getFirstPropertyValue('uid').toString() != "B39B2EDE-29B1-4118-8ED1-B1966F9209D2");
    }catch(e){
      return true;
    }
  });
```

**Reply by tkuiCO2** [reply]
Unfortunately neither of them work. Just to make sure, I added the snippet to helpers.gs line 184, saved it, went to code.gs and ran startsync (I also tried the uninstall-install method before startsync - same result), went to my Google calendar, refreshed it, and I expected to see this event gone… is that correct?

![CleanShot 2024-01-23 at 11 12 53](https://github.com/derekantrican/GAS-ICS-Sync/assets/106739694/5dd50ee8-bbe4-4a1c-9596-cac5d327b802)

**Reply by jonas0b1011001** [reply]
Move it right above `return result;`

**Reply by tkuiCO2** [reply, marked as answer]
Good News! Issue resolved! 

What I did is the following (before I saw your last answer).

1. Updated all code (all 4 files) to the latest - was few versions back.
2. added the 1st snippet (attached below) to helpers.gs in line 254 changing 'cn' to 'email' and setting my iCloud email address.
4. Saved
5. Uninstalled script
6. Deleted the calendar from Google Calendar
7. Installed the script
8. Ran StartSync

and voila! the iCal calendar was added to Google without this annoying series! 

Thanks for the help!

![CleanShot 2024-01-23 at 13 19 27](https://github.com/derekantrican/GAS-ICS-Sync/assets/106739694/be844d6c-a51c-45da-8921-502c719eaabc)
